### ISSUE TITLE
Problem in executing hive query

### ISSUE THREAD

**Issue by pidubey** [Esri/spatial-framework-for-hadoop#133]
I have faced problem creating table counties
CREATE TABLE counties (Area string, Perimeter string, State string, County string, Name string, BoundaryShape binary) 
> ROW FORMAT SERDE 'com.esri.hadoop.hive.serde.EsriJsonSerDe'
> STORED AS INPUTFORMAT 'com.esri.json.hadoop.EnclosedEsriJsonInputFormat'
> OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'; 

Error:
FAILED: SemanticException Cannot find class 'com.esri.json.hadoop.EnclosedEsriJsonInputFormat'

I have tried some solutions from online help but did't get succeded.

kindly let me know the issue.

Thanks with regards

**Comment by randallwhitman** [comment]
Did you `add jar`?
What version of Hive?

**Comment by pidubey** [comment]
Yes I have added jar

Ubuntu OS 16.04
Hadoop 2.6.0
Hive 2.3.0

I have created a directory in hdfs, copied all the jars in the folder and
then added jars using Hive. The jars are also in lib and  folder of Hive
and bash file is having class path of lib folder of Hive. The hadoop user
and group is having rights on both the folder but still I am getting the
error.

With regards
Piyush

**Comment by randallwhitman** [comment]
[For Hive-2.3, you'll need v2 of Spatial Framework](https://github.com/Esri/spatial-framework-for-hadoop/wiki/ST_Geometry-for-Hive-Compatibility-with-Hive-Versions), which is not yet released, so it will be necessary to build from source.

For the purposes of troubleshooting, are you able to create a table with a geometry, using a different input format?
```
1505,POINT(15 5)
0535,POINT(5 35)
2323,POINT(23 23)
3222,POINT(32 22)
3728,POINT(37 28)
2233,POINT(22 33)
2838,POINT(28 38)
3434,POINT(34 34)
6219,POINT(62 19)
7114,POINT(71 14)
7525,POINT(75 25)
6535,POINT(65 35)
5549,POINT(55 49)
6545,POINT(65 45)
4566,POINT(45 66)
```
```
CREATE TABLE `test15csv`(
  `rowid` int, 
  `wkt` string)
ROW FORMAT DELIMITED 
  FIELDS TERMINATED BY ',' 
STORED AS INPUTFORMAT 
  'org.apache.hadoop.mapred.TextInputFormat' 
OUTPUTFORMAT 
  'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'
```
`select rowid, ST_AsText(ST_GeomFromText(wkt)) from test15csv;`

**Comment by pidubey** [comment]
Dear Randall

           Thanks for the information.  Can u pls let me know the softwares
and version requirement to build V2 spatial framework from source. Current
spatial framework jar works with which version of Hive.

Thanks with regards

Piyush

**Comment by randallwhitman** [comment]
Build with maven and JDK as per https://github.com/Esri/spatial-framework-for-hadoop/blob/master/README.md .
I'd expect [Spatial Framework v1.2](https://github.com/Esri/spatial-framework-for-hadoop/releases/tag/v1.2.0) to [work with Hive 0.10 .. 2.1](https://github.com/Esri/spatial-framework-for-hadoop/wiki/ST_Geometry-for-Hive-Compatibility-with-Hive-Versions).

**Comment by pidubey** [comment]
Thanks for the information. Have a good day.

**Comment by randallwhitman** [comment]
I'd expect recently-released [Spatial-Framework v2.0](https://github.com/Esri/spatial-framework-for-hadoop/releases/tag/v2.0.0) to [work with Hive 0.11 .. 2.3](https://github.com/Esri/spatial-framework-for-hadoop/wiki/ST_Geometry-for-Hive-Compatibility-with-Hive-Versions).
Note: if compiling from source, you can try to use maven profiles to match the versions of Hive & Hadoop `-Phive-2.3 -Phadoop-2.6`.

**Comment by randallwhitman** [comment]
Note also 'com.esri.json.hadoop.EnclosedEsriJsonInputFormat' requires v1.2+ ; with v1.[01] only the older name would be available ~ #96.
